### PR TITLE
Add job cache test

### PR DESCRIPTION
**Pull request by Ch3LL**
### What does this PR do?

Add tests for removing and adding jobs in the cache. Also adds the returners directory to salts test suite. 
### Tests written?

Yes

### CODE REVIEW

**Comment by cachedout** [comment]
@Ch3LL Some lint, please: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/4014/violations/file/tests/integration/returners/local_cache.py/

**Comment by Ch3LL** [comment]
@cachedout k fixed it thanks

**Comment by rallytime** [comment]
@Ch3LL I think there maybe a returners reference missing from the changes you made to runtests.py. It doesn't appear that the jenkins test run picked up the new returner integration tests you've written. Mind taking another look?

**Comment by Ch3LL** [comment]
@rallytime okay I think I got all the options. where did you see it not running the test so I can make sure to check that in the future?

**Comment by rallytime** [comment]
@Ch3LL Hm. Looks like jenkins is not so happy right now, so we'll have to run these again when he's feeling better.

To see where the tests get picked up, just look at either the Ubuntu or CentOS links in the last successful test run here, click `Test Result` and then try to find the new tests on that page. The only reference to `returners` is for `unit.returners` tests, but if your tests were picked up, there would be a new `integration.returners.x` entry on that page.

**Comment by Ch3LL** [comment]
Okay great I'll take a look once the tests are re-run when jenkins is healthy again. thanks

**Comment by cachedout** [comment]
@Ch3LL Please let me know when this is ready to go.

**Comment by rallytime** [comment]
Go Go Jenkins!

**Comment by Ch3LL** [comment]
Go Go Jenkins!

**Comment by rallytime** [comment]
@cachedout I spoke with @Ch3LL offline and these are ready to go. This could probably use your expert review as well, but this is very nice and these tests will be excellent to have.

File: tests/integration/returners/local_cache.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+from salttesting.helpers import ensure_in_syspath
+ensure_in_syspath('../../')
+
+# Import salt libs
+import integration
+import salt.utils.job
+from salt.returners import local_cache
+
+log = logging.getLogger(__name__)
+if platform.uname()[0] == 'Darwin':
```
Comment (cachedout) [inline tests/integration/returners/local_cache.py]: Much this logic is available from `salt.integration` already. i.e. `salt.integration.TMP`.

**Comment by cachedout** [comment]
Just one comment in-line. Thanks, @Ch3LL and @rallytime

**Comment by cachedout** [comment]
Solid. This is great, @Ch3LL. Very much needed.

**Pull request merged**